Sources_v3/Workers/MessageUpdater.swift

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+class MessageUpdater<ExtraData: ExtraDataTypes>: Worker {
+    func getMessage(cid: ChannelId, messageId: MessageId, completion: ((Error?) -> Void)? = nil) {
+        let endpoint: Endpoint<MessagePayload<ExtraData>> = .getMessage(messageId: messageId)
+        apiClient.request(endpoint: endpoint) {
+            switch $0 {
+            case let .success(message):
+                self.database.write({ session in
+                    try session.saveMessage(payload: message, for: cid)
+                }, completion: { error in
+                    completion?(error)
+                })
+            case let .failure(error):
+                completion?(error)
+            }
+        }
+    }
+    
+    func deleteMessage(messageId: MessageId, completion: ((Error?) -> Void)? = nil) {
+        var shouldDeleteOnBackend = true
+        
+        database.write({ session in
+            guard
+                let dto = session.message(id: messageId), [.pendingSend, .sendingFailed].contains(dto.localMessageState)
+            else { return }
```

Comment:
This is a very critical piece of code. Can we make it a tiny bit more expressive? If I understand it correctly, what is happening here is: 
```
if <message exists only locally> {
  delete it and set shouldDeleteOnBackend to false
}
```
I'd like the actual code to read the same as the pseudo-code above :)